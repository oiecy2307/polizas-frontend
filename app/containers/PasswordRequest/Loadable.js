/**
 * Asynchronously loads the component for PasswordRequest
 */

import loadable from 'utils/loadable';

export default loadable(() => import('./index'));
