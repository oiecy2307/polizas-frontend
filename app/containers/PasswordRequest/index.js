/*
 * PasswordRequest
 *
 * This is the first thing users see of our App, at the '/' route
 *
 */

import React, { useState, useEffect, useContext } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { compose } from 'redux';
import { wsRequestRecovery } from 'services/auth';
import { aOpenSnackbar } from 'containers/App/actions';
import { getToken } from 'utils/helper';
import { GlobalValuesContext } from 'contexts/global-values';
import { isEmail } from 'validator';

import Input from 'components/InputText';
import LayersIcon from '@material-ui/icons/Layers';
import { SpaceBetween, Button } from 'utils/globalStyledComponents';
import {
  MainContainer,
  FormSection,
  Logo,
  H1,
  P,
  Form,
} from './styledComponents';
import getMessages from './messages';

function PasswordRequest({ history, dispatch }) {
  const { language } = useContext(GlobalValuesContext);
  const [messages] = useState(getMessages(language));
  const [username, setUsername] = useState('');

  useEffect(() => {
    evaluateToken();
  }, []);

  async function evaluateToken() {
    const token = await getToken();
    if (token) history.push('/');
  }

  async function handleRequestPasswordChange() {
    try {
      if (username.trim()) {
        const response = await wsRequestRecovery({ email: username.trim() });
        if (response.error) {
          dispatch(aOpenSnackbar('No se pudo enviar la solicitud', 'error'));
        } else {
          dispatch(aOpenSnackbar('Solicitud enviada a tu correo', 'success'));
        }
      }
    } catch (e) {
      dispatch(aOpenSnackbar('No se pudo enviar la solicitud', 'error'));
    }
  }

  const disabled = !isEmail(username);

  return (
    <MainContainer>
      <FormSection>
        <Logo>
          <LayersIcon style={{ color: '#108043', fontSize: 32 }} />
        </Logo>
        <H1>{messages.welcome}</H1>
        <P>{messages.welcomeMessage}</P>
        <Form>
          <Input
            label={messages.email}
            type="email"
            error={false}
            onChange={e => setUsername(e.target.value)}
            maxLength="150"
          />
          <SpaceBetween>
            <span />
            <Button disabled={disabled} onClick={handleRequestPasswordChange}>
              {messages.login}
            </Button>
          </SpaceBetween>
        </Form>
      </FormSection>
    </MainContainer>
  );
}

PasswordRequest.propTypes = {
  history: PropTypes.object.isRequired,
  dispatch: PropTypes.func.isRequired,
};

function mapDispatchToProps(dispatch) {
  return {
    dispatch,
  };
}

const withConnect = connect(mapDispatchToProps);

export default compose(withConnect)(PasswordRequest);
