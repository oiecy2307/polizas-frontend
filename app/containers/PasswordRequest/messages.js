/*
 * PasswordRequest Messages
 *
 * This contains all the text for the PasswordRequest container.
 */
import { defineMessages } from 'react-intl';

export const scope = 'app.containers.PasswordRequest';

export default language => {
  const values = {
    welcome: {
      es: 'Bienvenido a Suppdesk',
      en: 'Welcome to Suppdesk',
    },
    welcomeMessage: {
      es:
        'Ingresa tu dirección de correo y te enviaremos un correo con el link para recuperar tu contraseña.',
      en:
        "Write your email account and we'll send you an email with a link to recover your password",
    },
    email: {
      es: 'Correo',
      en: 'Email',
    },
    password: {
      es: 'Contraseña',
      en: 'Password',
    },
    login: {
      es: 'Enviar correo',
      en: 'Send email',
    },
  };
  return defineMessages({
    header: {
      id: `${scope}.header`,
      defaultMessage: 'This is the Users container!',
    },
    welcome: values.welcome[language],
    welcomeMessage: values.welcomeMessage[language],
    email: values.email[language],
    password: values.password[language],
    login: values.login[language],
  });
};
