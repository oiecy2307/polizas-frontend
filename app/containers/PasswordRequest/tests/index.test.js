import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';

import PasswordRequest from '../index';

describe('<PasswordRequest />', () => {
  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <IntlProvider locale="en">
        <PasswordRequest />
      </IntlProvider>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
