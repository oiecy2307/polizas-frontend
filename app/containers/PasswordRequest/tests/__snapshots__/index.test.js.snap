// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PasswordRequest /> should render and match the snapshot 1`] = `
<h1>
  <span>
    This is the PasswordRequest container!
  </span>
</h1>
`;
